tests/testthat/test-redundant_equals_linter.R

```diff
@@ -0,0 +1,30 @@
+test_that("redundant_equals_linter skips allowed usages", {
+  # comparisons to non-logical constants
+  expect_lint("x == 1", NULL, redundant_equals_linter())
+  # comparison to TRUE as a string
+  expect_lint("x != 'TRUE'", NULL, redundant_equals_linter())
+})
+
+test_that("redundant_equals_linter blocks simple disallowed usages", {
+  linter <- redundant_equals_linter()
+  eq_msg <- rex::rex("Using == on a logical vector is redundant.")
+  ne_msg <- rex::rex("Using != on a logical vector is redundant.")
+  expect_lint("x == TRUE", eq_msg, linter)
+  expect_lint("x != TRUE", ne_msg, linter)
+  expect_lint("x == FALSE", eq_msg, linter)
+  expect_lint("x != FALSE", ne_msg, linter)
+
+  # order doesn't matter
+  expect_lint("TRUE == x", eq_msg, linter)
```

Comment:
For the sake of completeness:

```suggestion
  expect_lint("TRUE == x", eq_msg, linter)
  expect_lint("TRUE != x", eq_msg, linter)
```

Reply:
I'll do this with patrick if we're going to be exhaustive :)